packages/react-native/Libraries/promiseRejectionTrackingOptions.js

```diff
@@ -22,7 +25,28 @@ let rejectionTrackingOptions: $NonMaybeType<Parameters<enable>[0]> = {
       // $FlowFixMe[method-unbinding] added when improving typing for this parameters
       message = Error.prototype.toString.call(rejection);
       const error: Error = (rejection: $FlowFixMe);
-      stack = error.stack;
+
+      const warning =
+        `Possible Unhandled Promise Rejection (id: ${id}):\n` +
+        `${message ?? ''}\n` +
+        (stack == null ? '' : stack);
+
+      // Print pretty unhandled rejections while on DEV
+      if (__DEV__) {
+        const parsedStack = parseErrorStack(error.stack);
+
+        LogBox.addLog({
+          level: 'warn',
+          message: {content: warning, substitutions: []},
+          componentStack: [],
+          stack: parsedStack,
+          category: 'possible_unhandled_promise_rejection',
+        });
```

Comment:
Can we avoid parsing the error stack here, and leave that op to LogBox?

Consider whether `addException` may be better here, since it already supports passing in a stack?

Reply:
I just tried but addException tackes an `Array<StackFrame>` on the `stack` property not a string. Tried modifying that but getting another error:

![Oscar Franco Screen 000116](https://github.com/facebook/react-native/assets/1634213/8534e334-c2a1-40dc-9e8b-6940fb17af95)

So it's a chicken and egg problem trying to use that. I can modify addLog to check for a string stack and parsing it there though, would that be a good compromise?